### DISCUSSION TITLE
Open Router Support

### DISCUSSION THREAD

**Discussion by f3smiley** [alagrede/znote-app#88] [Ideas]
Please add support for open router.

**Comment by alagrede** [comment]
Hi @f3smiley,  

Thank you for introducing me to this SaaS. It looks promising! From what I can see, it appears to be compatible with the OpenAI API already supported in Znote. You could potentially start using it with the **Ollama plugin** by tweaking or copy the default `useLocalAI` function (see the screenshot below).  

I'll plan some tests to explore integrating this service directly with Znote's internal AI tools, such as the **Chat View** and **AI Modification Tools**.  

Thanks again for your valuable feedback!  

![localLLM](https://github.com/user-attachments/assets/ffcb95c5-2f45-4ac4-a1dd-a6362c3614d3)

**Comment by alagrede** [comment]
OpenRouter looks fantastic, but it currently doesn’t provide an embeddings API. I’ll need to explore ways to work around this while keeping the installation process simple. I’ll get back to you with some potential solutions. Let’s stay in touch, and thank you again for pointing out this service!

**Comment by alagrede** [comment]
Hi @f3smiley 👋,  

I’ve made a quick update to add **OpenRouter** as an LLM provider for Znote's internal AI tools. Additionally, I’ve updated the [Ollama plugin tutorial](https://recipe.znote.io/recipes/ollama_plugin_tutorial) to demonstrate how to interact with OpenRouter using code.  

As mentioned yesterday, OpenRouter currently doesn’t support embeddings, which are crucial for improving response accuracy in the ChatView. For now, I’m relying on Ollama for embeddings. 

This integration is still experimental, so feel free to share your feedback or let me know if you need further adjustments or support.
